mempool/clist_mempool.go

```diff
@@ -543,7 +552,29 @@ func (mem *CListMempool) resCbRecheck(req *abci.Request, res *abci.Response) {
 	switch r := res.Value.(type) {
 	case *abci.Response_CheckTx:
 		if r.CheckTx.Code == abci.CodeTypeOK {
-			// Good, nothing to do.
+			tx := req.GetCheckTx().Tx
+			txHash := TxKey(tx)
+			e, ok := mem.txsMap.Load(txHash)
+			if !ok {
+				panic(fmt.Sprintf("Unexpected tx response from proxy during recheck\ntxHash=%s, tx=%X", txHash, tx))
+			}
+			celem := e.(*clist.CElement)
+			var postCheckErr error
+			if mem.postCheck != nil {
+				postCheckErr = mem.postCheck(tx, r.CheckTx)
```

Comment:
Would you modify calling `postCheck` first after this `switch` statement like tendermint? Because this change makes complex.

* https://github.com/tendermint/tendermint/blob/main/mempool/v0/clist_mempool.go#L378-L384
```
switch r := res.Value.(type) {
	case *abci.Response_CheckTx:
		var postCheckErr error
		if mem.postCheck != nil {
			postCheckErr = mem.postCheck(tx, r.CheckTx)
		}
		if (r.CheckTx.Code == abci.CodeTypeOK) && postCheckErr == nil {
```

Reply:
I like to use `postCheck` in current position, because `postCheckErr` is used in only the case `r.CheckTx.Code == abci.CodeTypeOK`. But current conditional branch is complex, so I tried to make code simple using early returns.